src/input.rs

```diff
@@ -0,0 +1,492 @@
+#![allow(clippy::too_many_arguments, clippy::type_complexity)]
+
+use std::time::Duration;
+
+use bevy::{
+    input::mouse::{MouseScrollUnit, MouseWheel},
+    prelude::*,
+    window::PrimaryWindow,
+};
+use cosmic_text::{Action, AttrsList, BufferLine, Cursor, Edit, Shaping};
+
+use crate::{
+    get_node_cursor_pos, get_timestamp, get_x_offset_center, get_y_offset_center,
+    save_edit_history, CosmicAttrs, CosmicEditHistory, CosmicEditor, CosmicFontSystem,
+    CosmicMaxChars, CosmicMaxLines, CosmicTextChanged, CosmicTextPosition, Focus, ReadOnly,
+    XOffset,
+};
+
+pub(crate) fn input_mouse(
+    windows: Query<&Window, With<PrimaryWindow>>, // Mouse
+    active_editor: Res<Focus>,                    // Both
+    keys: Res<Input<KeyCode>>,                    // Both
+    buttons: Res<Input<MouseButton>>,             // Mouse
+    mut cosmic_edit_query: Query<
+        (
+            &mut CosmicEditor,   // Both
+            &GlobalTransform,    // Mouse
+            &CosmicTextPosition, // Mouse, to determine point
+            Entity,              // Both
+            &XOffset,            // Mouse
+            Option<&mut Node>,
+            Option<&mut Sprite>,
+        ),
+        Without<ReadOnly>,
+    >,
+    mut font_system: ResMut<CosmicFontSystem>,    // Both
+    mut scroll_evr: EventReader<MouseWheel>,      // Mouse
+    camera_q: Query<(&Camera, &GlobalTransform)>, // Mouse
+) {
+    if active_editor.0.is_none() {
+        return;
+    }
+
+    let primary_window = windows.single();
+    let scale_factor = primary_window.scale_factor() as f32;
+    let (camera, camera_transform) = camera_q.iter().find(|(c, _)| c.is_active).unwrap();
+    for (mut editor, node_transform, text_position, entity, x_offset, node_opt, sprite_opt) in
+        &mut cosmic_edit_query.iter_mut()
+    {
+        if active_editor.0 != Some(entity) {
```

Comment:
User still should be able to select text on non-active editors.

Reply:
Ah yep I'll re-add the readonly bits